### PR TITLE
[SON-134] Weighted multisignature wallet integration

### PR DESCRIPTION
**Pull request by gladcow**
Allows to store bitcoin private keys only in SON config file, don't use bitcoind to create and sign transaction. I've tested locally primary wallet funds transfer during son wallet changes,  it works correctly.

### CODE REVIEW

File: programs/witness_node/genesis.json [hunk 1]
```diff
@@ -333,7 +333,7 @@
       "scale": 10000
     },
     "block_interval": 3,
-    "maintenance_interval": 86400,
+    "maintenance_interval": 120,
     "maintenance_skip_slots": 3,
```
Comment (obucina) [inline programs/witness_node/genesis.json]: Remove this file from PR

File: libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp [hunk 2]
```diff
@@ -480,13 +483,13 @@ std::vector<btc_txout> bitcoin_rpc_client::listunspent() {
 std::vector<btc_txout> bitcoin_rpc_client::listunspent_by_address_and_amount(const std::string &address, double minimum_amount) {
    std::string body = std::string("{\"jsonrpc\": \"1.0\", \"id\":\"pp_plugin\", \"method\": "
                                   "\"listunspent\", \"params\": [");
-   body += std::string("1,999999,[\"");
+   body += std::string("0,999999,[\"");
```
Comment (obucina) [inline libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp]: We cant have 0 here. We actually need like 5 or 6. Transactions with 0 confirmations might not exists when the block is mined.

File: libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp [hunk 3]
```diff
@@ -703,6 +706,38 @@ bool bitcoin_rpc_client::walletpassphrase(const std::string &passphrase, uint32_
    return false;
 }
 
+uint64_t bitcoin_rpc_client::outputamount(const std::string &txid, uint32_t output_num)
+{
```
Comment (obucina) [inline libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp]: Why do we have this one?

File: libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp [hunk 4]
```diff
@@ -852,86 +894,105 @@ void sidechain_net_handler_bitcoin::recreate_primary_wallet() {
...
          vector<string> son_pubkeys_bitcoin;
+         vector<uint64_t> son_votes;
          for (const son_info &si : active_sons) {
             son_pubkeys_bitcoin.push_back(si.sidechain_public_keys.at(sidechain_type::bitcoin));
+            son_votes.push_back(si.weight);
          }
 
-         if (!wallet_password.empty()) {
-            bitcoin_client->walletpassphrase(wallet_password, 5);
+         std::string full_address_info = create_multisig_address(son_pubkeys_bitcoin, son_votes);
```
Comment (obucina) [inline libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp]: The only change in this function should be the call to address creation function, and eventually call to transaction creation. Looks like most of existing changes with fixes are reverted. There are two calls to create_transaction. Refactor this function, to match the one from current SONs-base.

```
          auto active_sons = gpo.active_sons;
          vector<string> son_pubkeys_bitcoin;
+         vector<uint64_t> son_weights;
          for (const son_info &si : active_sons) {
             son_pubkeys_bitcoin.push_back(si.sidechain_public_keys.at(sidechain_type::bitcoin));
+            son_weights.push_back(si.weight);
          }
 
-         if (!wallet_password.empty()) {
-            bitcoin_client->walletpassphrase(wallet_password, 5);
-         }
-         uint32_t nrequired = son_pubkeys_bitcoin.size() * 2 / 3 + 1;
-         string reply_str = bitcoin_client->addmultisigaddress(nrequired, son_pubkeys_bitcoin);
+         string address_info = create_multisig_address(son_pubkeys_bitcoin, son_weights);
 
-         std::stringstream active_pw_ss(reply_str);
+         std::stringstream active_pw_ss(address_info);
          boost::property_tree::ptree active_pw_pt;
          boost::property_tree::read_json(active_pw_ss, active_pw_pt);
          if (active_pw_pt.count("error") && active_pw_pt.get_child("error").empty()) {
```

File: libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp (hunk 3 above)
Comment (gladcow) [reply libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp]: I need exact input amount in satoshis for signing. This amount can't restored from double value we use in other places,  so I query it from bitcoin node

File: libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp (hunk 4 above)
Comment (gladcow) [reply libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp]: Working with walletm  calculating nrequired and parsing jsob strings are psbt-specific actions,  so I've moved them to psbt-specific calls.

**Pull request closed without merging**